src/WebApi/Infrastructure/Middleware/BasicAuthMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using System.Net.Http.Headers;
+
+namespace Sindibad.SAD.WebTemplate.WebApi.Infrastructure.Middleware
+{
+    public class BasicAuthMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private const string Realm = "SindibadAPI";
+
+        public BasicAuthMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (!context.Request.Path.Value.Contains("api"))
+            {
+                await _next(context);
+                return;
+            }
+
+            if (!context.Request.Headers.ContainsKey("Authorization"))
+            {
+                context.Response.Headers["WWW-Authenticate"] = $"Basic realm=\"{Realm}\"";
+                context.Response.StatusCode = StatusCodes.Status403Forbidden;
+                await context.Response.WriteAsync("Access Forbidden.");
```

Comment:
I don't think we need to write anything in the response. Honestly, it's usually best practice just to return an empty 403. the status code is pretty self-explanatory